functions/functionsv2/firebase/hellofirestore/hellofirestore.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_cloudevent_firebase_firestore]
+
+// Package hellofirestore contains a Cloud Event Function triggered by a Cloud Firestore event.
+package hellofirestore
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/GoogleCloudPlatform/functions-framework-go/functions"
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/googleapis/google-cloudevents-go/cloud/firestoredata"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+func init() {
+	functions.CloudEvent("helloFirestore", HelloFirestore)
+}
+
+// HelloFirestore is triggered by a change to a Firestore document.
+func HelloFirestore(ctx context.Context, event event.Event) error {
+	dataAny := &anypb.Any{}
+	if err := proto.Unmarshal(event.Data(), dataAny); err != nil {
+		log.Fatal("proto.Unmarshal: ", err)
+	}
+
+	data := &firestoredata.DocumentEventData{}
```

Comment:
quibble: I think you need only declare this using `var`.

```
var data &firestoredata.DocumentEventData
```

Reply:
Not sure if that is valid. I could do
```
var data *firestoredata....
```
But then it would not be initialized If I am not mistaken.

Reply:
It initializes to the nil value, which is the same as what you've initialized it to here.